packages/gatsby/src/schema/schema.js

```diff
@@ -785,12 +789,34 @@ const addThirdPartySchemas = ({
         type.name !== `Date` &&
         type.name !== `JSON`
       ) {
+        const isCompositeType =
+          type instanceof GraphQLObjectType ||
+          type instanceof GraphQLInterfaceType
```

Comment:
Do we need to handle `GraphQLInputObjectType`? Was just checking https://github.com/graphql/graphql-js/blob/00ef21739731d1ea7fc61e4294b5bec5a2e535f2/src/type/definition.js#L1471 for type ... types that make use of `fields` but I lack context wether same (or similar) problem might apply to input type

Reply:
We need this fix only for types that may carry `Query` type in one of their fields (that's the only offender). Inputs cannot contain this type.